### DISCUSSION TITLE
add a lua compatibility layer for lower verison redis

### DISCUSSION THREAD

**Discussion by okhowang** [redis/go-redis#1765] [Ideas]
Can we add a mechanism for go-redis to implement as below?
```golang
client.SetNx(ctx, key, value, -1) 
/*
 send `SET key value NX` when redis version > 2.6.12
 or send `EVAL someScript 1 key value` when redis version < 2.6
*/
```

**Comment by vmihailenco** [comment]
We could provide some API to check Redis version, but I don't think we should provide Lua compatibility layer. It is a lot of work and I doubt we have resources for that.

**Reply by okhowang** [reply]
Lua compatibility layer could be contributed by community, but we have no API to do that.
`Hook` can not modify command itself.
the only way is wrap all go-redis API, it's a much large work

**Reply by vmihailenco** [reply]
Well, I guess we could merge such a contribution to the main repo if the code is clean and has tests.
